components/script/dom/xmlhttprequest.rs

```diff
@@ -1204,6 +1204,7 @@ impl XMLHttpRequest {
         Document::new(win,
                       None,
                       parsed_url,
+                      doc.origin().alias(),
```

Comment:
Given that the XHR specification doesn't specify any origin, I believe this should be an opaque origin instead.